apps/backend/src/modules/resend/service.ts

```diff
@@ -0,0 +1,287 @@
+import { ReactNode } from 'react'
+import { Resend } from 'resend'
+
+import {
+  AbstractNotificationProviderService,
+  MedusaError
+} from '@medusajs/framework/utils'
+import { ProviderSendNotificationDTO } from '@medusajs/types'
+
+import { BuyerAccountCreatedEmailTemplate } from './email-templates/buyer-account-created'
+import { BuyerCancelOrderEmailTemplate } from './email-templates/buyer-cancel-order'
+import { BuyerNewOrderEmailTemplate } from './email-templates/buyer-new-order'
+import { BuyerOrderShippedEmailTemplate } from './email-templates/buyer-shipped-order'
+import { ForgotPasswordEmailTemplate } from './email-templates/forgot-password'
+import { VendorAccountApprovedEmailTemplate } from './email-templates/vendor-account-approved'
+import { VendorAccountRejectedEmailTemplate } from './email-templates/vendor-account-rejected'
+import { VendorAccountSubmissionEmailTemplate } from './email-templates/vendor-account-updates-submission'
+import { VendorCanceledOrderEmailTemplate } from './email-templates/vendor-canceled-order'
+import { VendorNewOrderEmailTemplate } from './email-templates/vendor-new-order'
+import { VendorOrderShippingEmailTemplate } from './email-templates/vendor-shipping-order'
+import { VendorTeamInviteEmailTemplate } from './email-templates/vendor-team-invite'
+import { ResendOptions } from './types/options'
+
+export enum ResendNotificationTemplates {
+  BUYER_ACCOUNT_CREATED = 'buyer-account-created',
+  FORGOT_PASSWORD = 'forgot-password',
+  BUYER_NEW_ORDER = 'buyer-new_order',
+  BUYER_CANCELED_ORDER = 'buyer-canceled-order',
+  BUYER_ORDER_SHIPPED = 'buyer-order-shipped',
+  VENDOR_FORGOT_PASSWORD = 'vendor-forgot-password',
+  VENDOR_ACCOUNT_UPDATES_SUBMISSION = 'vendor-account-updates-submission',
+  VENDOR_ACCOUNT_UPDATES_APPROVAL = 'vendor-account-updates-approval',
+  VENDOR_ACCOUNT_UPDATES_REJECTION = 'vendor-account-updates-rejection',
+  VENDOR_NEW_ORDER = 'vendor-new-order',
+  VENDOR_CANCELED_ORDER = 'vendor-canceled-order',
+  VENDOR_ORDER_SHIPPED = 'vendor-shipped-order',
+  VENDOR_TEAM_MEMBER_INVITATION = 'vendor-team-member-invitation'
+}
+
+export function validateModuleOptions(
+  options: Record<any, any>,
+  moduleName: string
+) {
+  for (const key in options) {
+    if (!options[key]) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `No ${key} was provided in the ${moduleName} options. Please add one.`
+      )
+    }
+  }
+}
+
+class ResendNotificationProviderService extends AbstractNotificationProviderService {
+  static identifier = 'resend-notification'
+  private resendClient: Resend
+  private options: ResendOptions
+
+  constructor(container, options: ResendOptions) {
+    super()
+    validateModuleOptions(options, 'resendNotificationProvider')
+    this.resendClient = new Resend(options.api_key)
+    this.options = options
+  }
+
+  private async sendMail(subject: string, body: ReactNode, toEmail: string) {
```

Comment:
i think it would be a lot better if we would have this provider like this

```ts
 async send(notification: NotificationTypes.ProviderSendNotificationDTO) {
    const { data, error } = await this.resendClient.emails.send({
      from: notification.from?.trim() || this.config_.from,
      to: notification.to,
      subject: notification.content?.subject,
      react: notification.content?.react
    })

    if (error) {
      throw new MedusaError(MedusaError.Types.UNEXPECTED_STATE, error.message)
    }

    if (!data) {
      throw new MedusaError(
        MedusaError.Types.UNEXPECTED_STATE,
        'No data returned by resend client'
      )
    }

    return data
  }
```

have all of the email templates in the `src/email-templates` folder

and then use it like this in the subscriber

```ts
const notificationModuleService: INotificationModuleService =
    container.resolve(Modules.NOTIFICATION)
    
await notificationModuleService.createNotifications({
    to: "test@gmail.com",
    from: "test@medusajs.com", 
    channel: "email",
     content: {
     react: VendorAccountRejectedEmailTemplate({ data: orderData }),
     }
    ],
   
```


we then would have a good typescript check and seperation between use case and provider